### DISCUSSION TITLE
Dev environment setting up clarification questions

### DISCUSSION THREAD

**Discussion by nslee333** [alveusgg/alveusgg#344] [Q&A] [answered]
I'm setting up a local dev env following this [guide](https://github.com/alveusgg/alveusgg#how-to-develop--getting-started) and I got it working but I wanted to clarify. you're supposed to run `pnpm install` at: `root`, `server` and `website` right? 

I also was following along and I used `openssl rand -base64 32` to generate a passphrase like `zZ5rY7pNGmkYmHWc+nG5esT/xGEfIg80mkdN13PAkjI=` which is 44 characters. 

I pasted that into DATA_ENCRYPTION_PASSPHRASE,  `pnpm run -r dev` gave me a `DATA_ENCRYPTION_PASSPHRASE: [ 'String must contain exactly 32 character(s)' ]` error. 

What I ended up doing was truncating the passphrase until it was 32 characters long and `pnpm run -r dev` worked. I don't have any experience with SSL stuff. Am I using `openssl rand -base64 32` correctly? Is my solution appropriate or is there a better way to do it?

**Comment by MattIPv4** [comment, marked as answer]
You should only need to run install at the root -- as this is a workspace, it'll install everything :)

Ah, for the keys that need a 32 char key, the command should be `openssl rand -base64 24`. What you did is fine though.

**Comment by nslee333** [comment]
Ah, interesting, I think I had an issue where it was missing babel or something in the website folder, so I went an ran pnpm install in there. I might go back and see if it was a mistake on my end.

Cool, in .env.example the example above `DATA_ENCRYPTION_PASSPHRASE` uses `openssl rand -base64 32` I'll change it to 24.
